report/3.Theory/2.RTS.tex

```diff
@@ -44,4 +44,42 @@ \subsubsection{Task-based scheduling}
 
 This scheduling method can work both with and without pre-emption of tasks, meaning that a running task can be set on hold when a higher priority task is ready to run.
 When using this method for scheduling, a task with a low priority is allowed to run if no other task with a higher priority is ready to run. 
-Given a low priority task is running and a higher priority task is marked as ready, the low priority task will be pre-empted in order to allow the higher priority to run.
\ No newline at end of file
+Given a low priority task is running and a higher priority task is marked as ready, the low priority task will be pre-empted in order to allow the higher priority to run.
+
+\section{Response time analysis}
+Response time analysis is an exact schedulability test for an fixed-priority  assignment scheme on a single-processor system. 
+The response time analysis allows for prediction of the worst-case response time for each task which depends on the interference due to execution of higher priority tasks. 
+The worst-case response times are then compared to with the corresponding tasks deadlines to evaluate whether all tasks meets their deadlines. 
+
+Doing execution of a task, the task can be pre-empted if a higher priority task is released. 
+For this reason the response time of lower priority tasks suffer a certain amount of interference from the higher priority tasks during their execution. 
+Therefore the worst-case response time $R_i$ of a task $\tau_i$ is computed as the sum of its computation time $C_i$ and the worst-case interference $I_i$ it experiences, this is represented as follows:
+\begin{equation}\label{eq:responsetime}
+R_i = C_i + I_i
+\end{equation}
+
+The contribution of each higher priority task to the overall worst-case interference has to be evaluated individually by considering the interference of a single task $\tau_j$ of higher priority than $\tau_i$.
+The exact number of releases can be computed by means of the ceiling function and since each release of $\tau_J$ will impose on $\tau_i$ an interference of $C_j$ the worst-case interference imposed on $tau_i$ by $\tau_j$ is as follows:
+\begin{equation}
+\ceil*{\frac{R_i}{T_j}} \cdot C_j
+\end{equation}
+
+Since all tasks with a higher priority has an impact on the worst-case interference of a lower priority task all the higher priority tasks interference needs to be taken into account thus resulting in the following equation:
+\begin{equation}
+I_i = \sum_{j\in hp(i)} \cdot \ceil*{\frac{R_i}{T_j}} \cdot C_j
+\end{equation}
+
+Then recalling \ref{eq:responsetime} it is seen we get the following recursive relation for the worst-case response time $R_i$ for $\tau_i$:
```

Comment:
"Then recalling ... it is seen we get" - Sounds funny, could it be rephrased?

Reply:
Skal der måske stå when?